projects/RabbitMQ.Client/client/impl/Frame.cs

```diff
@@ -42,127 +42,117 @@
 using System.Buffers;
 using System.IO;
 using System.Net.Sockets;
+using System.Runtime.CompilerServices;
 using System.Runtime.ExceptionServices;
 using System.Runtime.InteropServices;
 using RabbitMQ.Client.Exceptions;
-using RabbitMQ.Client.Framing;
 using RabbitMQ.Util;
 
 namespace RabbitMQ.Client.Impl
 {
-    class HeaderOutboundFrame : OutboundFrame
+    internal static class Framing
     {
-        private readonly ContentHeaderBase _header;
-        private readonly int _bodyLength;
-
-        internal HeaderOutboundFrame(int channel, ContentHeaderBase header, int bodyLength) : base(FrameType.FrameHeader, channel)
-        {
-            _header = header;
-            _bodyLength = bodyLength;
-        }
-
-        internal override int GetMinimumPayloadBufferSize()
-        {
-            // ProtocolClassId (2) + header (X bytes)
-            return 2 + _header.GetRequiredBufferSize();
-        }
-
-        internal override int WritePayload(Memory<byte> memory)
-        {
-            // write protocol class id (2 bytes)
-            NetworkOrderSerializer.WriteUInt16(memory.Span, _header.ProtocolClassId);
-            // write header (X bytes)
-            int bytesWritten = _header.WriteTo(memory.Slice(2), (ulong)_bodyLength);
-            return 2 + bytesWritten;
-        }
-    }
-
-    class BodySegmentOutboundFrame : OutboundFrame
-    {
-        private readonly ReadOnlyMemory<byte> _body;
-
-        internal BodySegmentOutboundFrame(int channel, ReadOnlyMemory<byte> bodySegment) : base(FrameType.FrameBody, channel)
-        {
-            _body = bodySegment;
-        }
-
-        internal override int GetMinimumPayloadBufferSize()
-        {
-            return _body.Length;
-        }
-
-        internal override int WritePayload(Memory<byte> memory)
+        /* +------------+---------+----------------+---------+------------------+
+         * | Frame Type | Channel | Payload length | Payload | Frame End Marker |
+         * +------------+---------+----------------+---------+------------------+
+         * | 1 byte     | 2 bytes | 4 bytes        | x bytes | 1 byte           |
+         * +------------+---------+----------------+---------+------------------+ */
```

Comment:
<3

Reply:
Took my way too long to figure this out from the code itself, thought that might help others better understand :)